### PR TITLE
Dump results to file while benchmark is on-going

### PR DESCRIPTION
**Pull request by jan-grzybek-ampere**
* Allows model zoo dev to dynamically adjust benchmarking time based on produced samples so far
* Unifies concept of batch size and variable input length as "task size" and "workload_size" (workload size being collection of singular task sizes) - no good reason to consider batch size and other adjustable dimension of input separately
* Runner base class for better modularity

### CODE REVIEW

File: utils/benchmark.py [hunk 1]
```diff
@@ -60,43 +62,123 @@ def get_intra_op_parallelism_threads():
...
+                    "finish_times": self._finish_times[self.warm_up_runs:times_invoked]
+                }, f)
+
+    def _dump_loop(self):
+        while self._do_dump:
+            time.sleep(5)
+            self._dump_results()
+
+    def abort_maybe(self):
+        if self._results_dir is not None and os.path.isfile(os.path.join(self._results_dir, "STOP")):
```
[the remark this answers is not in the document]
Comment (jan-grzybek-ampere) [reply utils/benchmark.py:111]: model zoo dev or by hand

File: utils/benchmark.py [hunk 2]
```diff
@@ -137,9 +218,11 @@ def run_model(single_pass_func, runner, dataset, batch_size, num_runs, timeout,
                     single_pass_func(runner, dataset)
                     timeout_pbar.n = int(min(time.time() - start, timeout))
                     timeout_pbar.refresh()
+                    runner.abort_maybe()
             else:
                 for _ in tqdm(range(num_runs)):
                     single_pass_func(runner, dataset)
+                    runner.abort_maybe()
```
[the remark this answers is not in the document]
Comment (jan-grzybek-ampere) [reply utils/benchmark.py:225]: if a file STOP has been created that means user wants the benchmarking to be immediately & gracefully terminated

File: utils/pytorch.py [hunk 3]
```diff
@@ -56,86 +55,72 @@ def __init__(self, model, disable_jit_freeze=False, example_inputs=None, func=No
...
-            self.__start_times, self.__finish_times, self.__times_invoked, batch_size,
-            variable_input_lengths=variable_input_lengths
-        )
+        return self.print_metrics()
 
 
-class PyTorchRunnerV2:
+class PyTorchRunnerV2(Runner):
     def __init__(self, model):
+        super().__init__()
```
[the remark this answers is not in the document]
Comment (jan-grzybek-ampere) [reply utils/pytorch.py:115]: that's right

File: utils/benchmark.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jan-grzybek-ampere) [reply utils/benchmark.py:111]: check the corresponding PR in model zoo dev to get bigger picture

**Review by dkupnicki — APPROVED** [review]

**Pull request merged**